### ISSUE TITLE
Transformable_value_error:  Error can not access tag

### ISSUE THREAD

**Issue by oscarfasanchez** [usgs/pestpp#197]
Hello!
I'm running Pestpp-GLM
and after calculating the FOSM-based Monte Carlo of the first iteration I got this error in the rec file
![image](https://user-images.githubusercontent.com/39230212/179429343-29e046a1-b505-4bf8-8e88-3b5085731161.png)
the log file shows this 
![image](https://user-images.githubusercontent.com/39230212/179429532-236bdd22-e152-4d06-85fe-36790ff45d45.png)

there are convergence errors as can be seen in the rmr file 
![image](https://user-images.githubusercontent.com/39230212/179429564-c29206ea-ba2d-4fa0-93dc-7078d7147c64.png)
and even in the  rec file shows problems with upgrade runs(but very few jacobian runs failed-3%)
![image](https://user-images.githubusercontent.com/39230212/179429664-23f99ca2-9e73-4ac8-af44-50b8655bccf0.png)

is this problem generated by the failed models? or it is caused by something else
files were written by Pyemu, using manually set pilot points
Thanks in advance for any help

**Comment by BJEANNOT0** [comment]
Hello,

here are my thoughts about your problem. I emphasize that I am just a user of Pest++, like you, and not the developper.

I would advise to change the values parval1,parlbnd and parubnd of your parameters, because it seems that you are using parameters that often do not enable your modele to converge (These could be because the parameters you chose bear no physical sense).

If you think the problem solely come from the FOSM analysis, you can try disabling it by setting the "uncertainty" control variable to false.

Also, I see that you are running Pest++ in parallel. I would advise that you set the control varibale "panther_agent_freeze_on_fail" to true so that each time an agent does not manage to finish its run, it instantly stops, so that you can investigate why the run failed. 

FInally, I see that your first image mentions something about "transformable value error". This makes me wonder if you are using a scale or an offset to your parameters ? If you do, be cautious if you use log-transformed parameters because you cannot transform a parameter that has a negative value.

Best regards,

**Comment by jtwhite79** [comment]
Its not unusual for the upgrade runs to struggle compared to the jacobian runs - the upgrade runs are when the parameters are really changing.  But that error is troubling.  Can you upload your pst file and rec file?

**Comment by oscarfasanchez** [comment]
Hello, @BJEANNOT0 , thanks for your insights, I think that uncertainty is not the problem, because  I used those parameters and bound parameters with IES without problems. I'll  set the control variable "panther_agent_freeze_on_fail" to investigate my convergence problems 👍.
Sure @jtwhite79 I'll attach them here
[model_pest.zip](https://github.com/usgs/pestpp/files/9141668/model_pest.zip)

**Comment by jtwhite79** [comment]
I think I see the problem.  The parameter that is missing in the transformable class is one that had its derivative run fail so its missing in the jco.  Lemme check if this is whats causing the error...

**Comment by jtwhite79** [comment]
Ok Ive dug into this a bit more with test problem and when I force all the lambdas to fail, the latest version of pestpp-glm exits more gracefully with an error message.  If you are wanting continue "as is", can you try the latest release (5.1.17 I think).

Conceptually tho, there are derivative runs that failed meaning the model didn't tolerate really small changes in parameters, and then every lambda run failed.  That seems like cause for concern RE model stability.  I might be tempted to try to understand why the model didnt tolerate small parameter increment changes...

**Comment by oscarfasanchez** [comment]
Thank you @jtwhite79 , I just tried to use the 5.1.17 version, to start the process again, but using the previous jacobian matrix to debug model failures, and I get a similar problem than before. 
![image](https://user-images.githubusercontent.com/39230212/179872054-a9dd41c8-f860-4eb3-b07d-8e85f5a4bd76.png)
due to the lack of 14 sensitivities (from 461) in the jacobian matrix, PEST++ stops with an error.  is there a way to continue?

**Comment by jtwhite79** [comment]
I think You could "fix" those missing pars in the control file as a quick work around....

**Comment by oscarfasanchez** [comment]
Ok, thanks;  I think it would be nice that Pest++ when you provide a jacobian matrix, it converts provisionally that values to zeros, or maybe that tries to calculate it by itself

**Comment by oscarfasanchez** [comment]
Hello @jtwhite79,  I have the same problem again with "Transformable_value_error:  Error can not access tag" with pest 5.1.20
this time I only had 1 lambda run fail, but 6% of derivatives failed
I think it is a problem with DERFORGIVE
![image](https://user-images.githubusercontent.com/39230212/187986448-6dba1d5c-e082-4821-995d-b93465ec5b59.png)

![image](https://user-images.githubusercontent.com/39230212/187986513-bd00665a-0d2c-4984-9025-6b5f8105c4bb.png)

I share my pst and rec file
[model_pest.zip](https://github.com/usgs/pestpp/files/9472405/model_pest.zip)

basically is the same model attached here by me after fixing the jacobian:
httattachedhub.com/usgs/pestpp/issues/198#issuecomment-1221616275

**Comment by jtwhite79** [comment]
@oscarfasanchez I'll take a look at this again asap.

**Comment by jtwhite79** [comment]
Looks like this is a bug from an interaction between `derforgive` and `glm_accept_mc_phi`.  Ive got a patch running now...

**Comment by oscarfasanchez** [comment]
I was wondering if this bug may be related to some weird results in linear uncertainty quantification. When I plot  a forecast I got this in the first iteration -model_pest.0.pred.usum.csv:
![image](https://user-images.githubusercontent.com/39230212/189203424-d9eb95b1-2127-43a4-8447-edd9dc819e7a.png)
but extremely high values in the next one -model_pest.1.pred.usum.csv
![image](https://user-images.githubusercontent.com/39230212/189203657-80160ad0-52a4-45e7-9141-206b4dfa9d39.png)

I guess that missing values in the Jacobian can lead to erratic results in the computing of the uncertainty.

**Comment by jtwhite79** [comment]
yeah it sounds like the model might a little unstable if derivative runs are failing, which means the derivatives themselves might be a little dodgy...

**Comment by oscarfasanchez** [comment]
so if I achieve converge in those failed models(28 of 461) and splice those two jacobians won't be of any help this time, right? this has nothing to do with the previous bug, but with the derivatives already calculated.

**Comment by jtwhite79** [comment]
correct - if you can fill a jco for the failed derivative runs and splice them together, you can use pyemu to get FOSM results...the iteration-level FOSM and MC results are just to provide you the option to ctrl+c at any time and still have some uncertainty estimates.
